### ISSUE TITLE
Investigate/fix secure messages sorting

### ISSUE THREAD

**Issue by kpethtel** [department-of-veterans-affairs/va-mobile-app#8536]
Per [slack conversation here](https://dsva.slack.com/archives/C018V2JCWRJ/p1714655452940929), users have reported that secure messages appear in non-chronological order. The messages index sorts according to a param provided by the front end:
`resource = resource.sort(params[:sort])`
As far as I can tell, the front end does not use that param. It's plausible that sorting by nil may be modifying the default ordering. Alternatively, it's possible the upstream is sending the messages in some other order. 

AC:
If reproducible:
- determine what the default ordering is from upstream
- determine whether sorting by nil is modifying the order. make code changes as necessary
- confirm with product that we want to sort chronologically by default and implement those changes if appropriate

If not reproducible:
- perhaps add logging
- check for any sorting that happens on the front end

**Comment by aherzberg** [comment]
No issue was found with the message index code so a log was added to try to find any issues with upstream data. No issues were found with the logged data either so I'm unable to reproduce this issue. The only path forward I see on this is that we get production access to log that specific user or FE does their own investigation to see if anything could happening on their end. I've already checked with them and for the message index, they just list it in the order we give them and for the thread index, they sort it themselves but did not find any issues with it. So a more in depth investigation ticket for FE may not be worth it. 

I've created a ticket to remove the logs I've added but want to give them some more time on the off chance it catches something. https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8714
